bittensor/utils/__init__.py

```diff
@@ -95,42 +98,43 @@ def solve_for_difficulty( block_hash, difficulty ):
             break
     return nonce, seal
 
-
-def solve_for_difficulty_fast( subtensor ):
+def solve_for_difficulty_fast( subtensor, num_processes: int = 5, update_interval: int = 100000 ) -> None:
+    """
+    Solves the POW for registration using multiprocessing.
+    Args:
+        subtensor
+            Subtensor to connect to for block information and to submit.
+        num_processes: int
+            Number of processes to use.
+        update_interval: int
+            Number of nonces to solve before updating block information.
+    Note: 
+    - We should modify the number of processes based on user input.
+    - We can also modify the update interval to do smaller blocks of work,
+        while still updating the block information after a different number of nonces,
+        to increase the transparency of the process while still keeping the speed.
+    """
     block_number = subtensor.get_current_block()
     difficulty = subtensor.difficulty
     block_hash = subtensor.substrate.get_block_hash( block_number )
     while block_hash == None:
         block_hash = subtensor.substrate.get_block_hash( block_number )
     block_bytes = block_hash.encode('utf-8')[2:]
     
-    meets = False
     nonce = -1
     limit = int(math.pow(2,256) - 1)
-    best = math.inf
-    update_interval = 100000
     start_time = time.time()
 
     console = bittensor.__console__
-    with console.status("Solving") as status:
-        while not meets:
-            nonce += 1 
-
-            # Create seal.
-            nonce_bytes = binascii.hexlify(nonce.to_bytes(8, 'little'))
-            pre_seal = nonce_bytes + block_bytes
-            seal = hashlib.sha256( bytearray(hex_bytes_to_u8_list(pre_seal)) ).digest()
-
-            seal_number = int.from_bytes(seal, "big")
-            product = seal_number * difficulty
-            if product - limit < best:
-                best = product - limit
-                best_seal = seal
-
-            if product < limit:
-                return nonce, block_number, block_hash, difficulty, seal
-
-            if nonce % update_interval == 0:
+    with console.status("Solving") as status:        
+        found_solution = multiprocessing.Value('q', -1, lock=False) # int
+        best = multiprocessing.Value(ctypes.c_char_p, lock=True) # byte array to get around int size of ctypes
+        best.raw = struct.pack("d", float('inf'))
+        best_seal = multiprocessing.Array('h', 32, lock=True) # short array should hold bytes (0, 256)
+        with multiprocessing.Pool(processes=num_processes, initializer=initProcess_, initargs=(solve_, found_solution, best, best_seal)) as pool:
+            while found_solution.value == -1: # no solution found
```

Comment:
Extra work, maybe not for this PR, but we should probably be checking to see if the wallet has been registered during this process. For instance, if another computer computing the registration solves it, then this should close down. However this would require adding this tool kit into the subtensor code base which is already a bit verbose. Food for thought.

Reply:
If we did do this, we can add the registration arguments into --subtensor. which would give it a good namespace.